### PR TITLE
chore: use `ASRUtils::use_experimental_simplifier` for enabling/disabling experimental simplifier

### PR DESCRIPTION
**Pull request by gxyd**
## Description

This will ensure that there is only a single place to enable/disable experimental simplifier pass, instead of use `pass_options.experimental_simplifier` at some places and `ASRUtils::use_experimental_simplifier` at others.

Also, toggling experimental_simplififer in hard-coded manner will produce expected results, once `ASRUtils::use_experimental_simplifier` is set to true/false, it will be used as that value throughout our codebase.

### CODE REVIEW

**Comment by gxyd** [comment]
To test that it's working correctly, I'll try to enable third party tests will *experimental simplifier* on.

**Comment by gxyd** [comment]
I haven't made changes to test the below third party packages with `--experimental-simplifier`:
* Test fpm (as it depends on `build.sh` file, which is present in a particular repository)
* Test fastGPT ( ubuntu-latest ) (I don't have access to an ubuntu distribution to make sure locally that things will work)

**Comment by gxyd** [comment]
Ah, the CI errors.

**Comment by gxyd** [comment]
Remaining #TODO

* Test fpm (as it depends on build.sh file, which is present in a particular repository)

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -591,6 +613,13 @@ jobs:
             $(pwd)/../src/bin/lfortran ./examples/example_lmdif1.f90 --legacy-array-sections
             $(pwd)/../src/bin/lfortran ./examples/example_lmder1.f90 --legacy-array-sections
 
+            # Test with experimental simplifier
+            $(pwd)/../src/bin/lfortran ./src/minpack.f90 -c --legacy-array-sections
```
Comment (certik) [inline .github/workflows/CI.yml:617]: This is missing `--experimental-simplifier`

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -629,6 +669,18 @@ jobs:
             ldd ./test_chat
             ldd ./test_more_inputs
 
+            # Test with experimental simplifier
+            git clean -dfx
+            FC=$(pwd)/../src/bin/lfortran CMAKE_PREFIX_PATH=$CONDA_PREFIX cmake -DFASTGPT_BLAS=OpenBLAS -DCMAKE_Fortran_FLAGS="--experimental-simplifier --realloc-lhs" .
```
Comment (gxyd) [inline .github/workflows/CI.yml:674]: `--realloc-lhs` is needed, see: https://github.com/lfortran/lfortran/pull/5182/files#diff-3ab46ee209a127470fce3c2cf106b1a1dbadbb929a4b5b13656a4bc4ce19c0b8R623 in *simplifier_pass* branch. Same for all the lines below as well, where `FC` is being assigned.

**Pull request merged**